src/modules/base.ts

```diff
@@ -1,119 +1,137 @@
 /**
  * Base module for all API modules
  *
- * @module modules/base
- *
  * @since 1.0.0
  */
 import commsUtils from '../utils/comms';
-import {name as PROTOCOL_AGENT_NAME, version as PROTOCOL_AGENT_VERSION} from '../../package.json';
 
 const PROTOCOL_NAME = 'purecloud-client-apps';
 
-const ACTION_NAME_KEY = 'action';
-const PROTOCOL_NAME_KEY = 'protocol';
-const PROTOCOL_AGENT_NAME_KEY = 'protocolAgentName';
-const PROTOCOL_AGENT_VERSION_KEY = 'protocolAgentVersion';
+interface ProtocolDetails {
+    name: string;
+    agentName: string;
+    agentVersion: string;
+}
+
+export interface SDKMessagePayload {
+    action: string;
+    hook?: string;
+    protocol: string;
+    protocolAgentName: string;
+    protocolAgentVersion: string;
+}
+
+export type MessageListener = (event: SDKMessagePayload) => void;
+export type MessagePayloadFilter = (payload: SDKMessagePayload) => boolean;
+
+type ListenerOptions = Partial<{
```

Comment:
It's not used currently by the lifecycle module (the only module utilizing listeners) but would we want to export this in case an extending class would need to reference the type?

Reply:
IMO no need to export until someone actually needs it